crates/containerd-shim-wasm/src/sys/unix/container/instance.rs

```diff
@@ -64,7 +65,7 @@ impl<E: Engine> SandboxInstance for Instance<E> {
         Ok(Self {
             id,
             exit_code: WaitableCell::new(),
-            rootdir,
+            container: RwLock::new(container),
```

Comment:
question: Why is a lock required for container?

Reply:
the operations we do on the container (start, kill, delete...) require mutable access, while we are only receiving an immutable reference. That's why we need the lock.

As to what we were doing before, the container object reads and write the state to disk. We were loading the container from disk. There's nothing enforcing synchronization at the filesystem level, but these operations should indeed be synchronized. With the previous version we could have had two mutable `Container` instances simultaneously trying to write different states to the same files.

Reply:
Is there a scenario where that would happen when this is driven by containerd?

Reply:
I don't know. @cpuguy83 , do you know the answer?

In any case, I would expect the lock to be cheaper than loading and parsing the state from disk.

Reply:
It looks like there is a pending question here. I agree with @jprendes that this PR is a net gain for the perf, so I am happy to merge this to the main tree.

Reply:
Its probably faster in the startup, maybe a bit more memory consumption 🤷